File: evals/fixtures/documented_constraint_violated/go/catch_redirect_breaks_origin.go
// edge/asset.go  (the diff adds the docs-component redirect case)
func (p *Proxy) Serve(key string, body []byte) Response {
	switch filepath.Ext(key) {
	case ".html", ".css", ".js", ".json", ".woff2":
		// Text assets we inline as base64 through the proxy. They must never be
		// returned as a redirect to object storage: a redirect changes the origin,
		// which breaks the relative links in .html/.css/.js and fails CORS for
		// .json/.woff2. Membership in this set means "must stay same-origin".
		encodedLen := (len(body) + 2) / 3 * 4
		// Large docs bundles exceed the inline cap; redirect them to a signed URL.
		if p.Component == "docs" && encodedLen > inlineLimit {
			return Response{Status: 307, Location: p.SignedURL(key)}
		}
		return Response{Status: 200, Body: base64Encode(body), ContentType: contentType(key)}
	default:
		return Response{Status: 307, Location: p.SignedURL(key)}
	}
}

File: evals/fixtures/documented_constraint_violated/python/catch_redirect_breaks_origin.py
# edge/asset_proxy.py  (the diff adds the `component == "docs"` redirect branch)
def serve_asset(self, key: str, body: bytes) -> Response:
    suffix = Path(key).suffix
    if suffix in (".html", ".css", ".js", ".json", ".woff2"):
        # These text assets are served base64-inline through the proxy. They must
        # never be handed back as a redirect to object storage: the redirect moves
        # them to a different origin, which breaks the relative links inside
        # .html/.css/.js and trips CORS for .json/.woff2. Every suffix in this set
        # depends on staying same-origin.
        headers = self.security_headers()
        encoded_len = (len(body) + 2) // 3 * 4
        # Large docs bundles blow past the inline size cap, so redirect those to a
        # signed object URL — docs bundles are self-contained libraries.
        if self.component == "docs" and encoded_len > INLINE_LIMIT:
            headers["location"] = self.signed_object_url(key)
            return Response(status=307, headers=headers)
        headers["content-type"] = self.content_type(key)
        return Response(status=200, body=b64encode(body), headers=headers)
    return Response(status=307, headers={"location": self.signed_object_url(key)})
